pkg/ebpf/uprobes/procfs.go

```diff
@@ -48,14 +54,26 @@ func waitUntilSucceeds[T any](p *ProcInfo, procFile string, readFunc func(string
 	var result T
 	err := errIterStart
 	end := time.Now().Add(procFSUpdateTimeout)
+	failedTimes := 0
 
 	for err != nil && end.After(time.Now()) {
 		result, err = readFunc(filePath)
 		if err != nil {
+			failedTimes++
 			time.Sleep(procFSUpdateInterval)
 		}
 	}
 
+	if err == nil {
+		if failedTimes > 0 {
+			readTemporaryFail.Add(int64(failedTimes))
+		}
```

Comment:
IIRC, `Add` verifies the input parameter is greater than 0, so you can simply remove if-clause

Reply:
Done.